components/Panel/DelegationPanel.tsx

```diff
@@ -0,0 +1,231 @@
+import { Button, LoadingSpinner, PanelErrorBanner, TextInput } from "components";
+import { getAddress, isAddress } from "helpers";
+import { useDelegationContext, useErrorContext, usePanelContext, useUserContext } from "hooks";
+import NextLink from "next/link";
+import One from "public/assets/icons/one.svg";
+import Three from "public/assets/icons/three.svg";
+import Time from "public/assets/icons/time-with-inner-circle.svg";
+import Two from "public/assets/icons/two.svg";
+import { useEffect, useState } from "react";
+import styled, { css } from "styled-components";
+import { PanelFooter } from "./PanelFooter";
+import { PanelTitle } from "./PanelTitle";
+import { PanelSectionText, PanelSectionTitle, PanelWrapper } from "./styles";
+
+export function DelegationPanel() {
+  const { closePanel } = usePanelContext();
+  const { address } = useUserContext();
+  const { addErrorMessage, clearErrorMessages } = useErrorContext("delegation");
+  const [delegateAddressToAdd, setDelegateAddressToAdd] = useState("");
+  const {
+    getDelegationStatus,
+    sendRequestToBeDelegate,
+    getPendingSentRequestsToBeDelegate,
+    getDelegationDataFetching,
+  } = useDelegationContext();
+
+  const delegationStatus = getDelegationStatus();
+
+  useEffect(() => {
+    // only show this panel when the user has not yet entered a delegation relationship, or the user has requested another wallet to be their delegate wallet
+    if (!(delegationStatus === "no-delegation" || delegationStatus === "delegator-pending")) {
```

Comment:
The only time the user can show this panel is when clicking "add wallet" on the wallet settings page. This button is only shown when the user has not requested another wallet to be their delegate. Hence we close the panel in other cases. The exception is that when user completes this add wallet flow and moves into the `delegator-pending` status, we don't want to close the panel and instead show the user information about the request they just made.